### PR TITLE
Add frame lock option for OccupancyGrid topics

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
Add a new topic settings modal for OccupancyGrid topics. The modal introduces a `Frame lock` option. When the `Frame lock` option is enabled, occupancy grids are placed within the 3d scene using transformations from the _current time_ rather than the `header.stamp` time as is the default behavior.

image

**Description**
Adding a frame lock option to occupancy grid topics allows users who publish occupancy grids as static maps to publish their message infrequently while still having the grid display correctly relative to other visual elements as new transforms are received.

Fixes: #3018

### CODE REVIEW

File: packages/studio-base/src/panels/ThreeDimensionalViz/SceneBuilder/index.ts [hunk 1]
```diff
@@ -197,6 +197,7 @@ function computeMarkerPose(
   }
   let srcTime: Time;
   let dstTime: Time;
+
```
Comment (defunctzombie) [inline packages/studio-base/src/panels/ThreeDimensionalViz/SceneBuilder/index.ts:200]:
```suggestion
```

File: packages/studio-base/src/panels/ThreeDimensionalViz/TopicSettingsEditor/GenericSettingsEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        control={
+          <Checkbox
+            checked={settings.frameLocked === true}
+            onChange={(_ev, checked) => onFieldChange("frameLocked", checked)}
+          />
+        }
+        label="Frame lock"
+      />
+      <SDescription>
+        By default, topics are translated into the 3d scene using their header stamp time. Enabling
```
Comment (jtbandes) [inline packages/studio-base/src/panels/ThreeDimensionalViz/TopicSettingsEditor/GenericSettingsEditor.tsx:41]:
```suggestion
        By default, topics are translated into the 3D scene using their header stamp time. Enabling
```

Comment (jtbandes) [reply packages/studio-base/src/panels/ThreeDimensionalViz/TopicSettingsEditor/GenericSettingsEditor.tsx:41]: Not sure I like this explanation overall. I think this is clearer: https://answers.ros.org/question/290293/what-exactly-does-frame_locked-mean/?answer=290337#post-id-290337

**Review by jtbandes — APPROVED** [review]
Long live the unholy union

**Pull request merged**